### PR TITLE
test: improve `commands-action-button.tsx` coverage

### PR DESCRIPTION
**Pull request by ckerr**
no functional changes.

This is a refactor of `tests/renderer/components/commands-publish-button-spec.tsx` that fills in some of the edge cases, e.g. the description dialog and some error handling.

It reduces copy-paste code duplication by nesting 'describe' sections with beforeEach() functions, e.g. in `describe('publish mode'` there is a beforeEach() that creates a publish button ready to be tested.

### CODE REVIEW

File: src/renderer/components/commands-action-button.tsx [hunk 1]
```diff
@@ -259,21 +259,15 @@ export class GistActionButton extends React.Component<
...
-          await this.handleDelete();
-          break;
-      }
-    } else {
-      await this.handlePublish();
+    if (gistId && actionType === GistActionType.update) {
+      return this.handleUpdate();
+    }
+
+    if (gistId && actionType == GistActionType.delete) {
```
Comment (erickzhao) [inline src/renderer/components/commands-action-button.tsx:266]: Is the `==` intentional here?

File: tests/renderer/components/commands-publish-button-spec.tsx [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+    delete: jest.fn(),
+    update: jest.fn(),
+  };
+}
+
+describe('Action button component', () => {
+  const description = 'Electron Fiddle Gist';
+  const errorMessage = '💀';
+  let app: AppMock;
+  let mocktokit: OctokitMock;
```
Comment (erickzhao) [inline tests/renderer/components/commands-publish-button-spec.tsx:38]: hehe `mocktokit`

File: src/renderer/components/commands-action-button.tsx (hunk 1 above)
Comment (ckerr) [reply src/renderer/components/commands-action-button.tsx]: Nope. Fixed 9a1a55bc95fe4085fb28fc1df2220ae6cbd5de64

File: src/renderer/components/commands-action-button.tsx [hunk 3]
```diff
@@ -259,21 +259,15 @@ export class GistActionButton extends React.Component<
...
-          break;
-      }
-    } else {
-      await this.handlePublish();
+    if (gistId && actionType === GistActionType.update) {
+      return this.handleUpdate();
+    }
+
+    if (gistId && actionType === GistActionType.delete) {
+      return this.handleDelete();
```
Comment (codebytere) [inline src/renderer/components/commands-action-button.tsx:267]: this pattern is a little confusing since there's no return value to speak of here - what motivated this change?

Comment (ckerr) [reply src/renderer/components/commands-action-button.tsx]: The main motivation for refactoring performGistAction() is to reduce the number of logic branches needed to reach the three possible outcomes of that function (publish, update, delete). However now that I reread it, I see three more branches can be removed while making the intent of the 'return' functions clearer. Revised in cff84ed.

The return values are the promises created by handlePublish(), handleDelete(), and handleDelete(). I don't feel super strongly about this, but my thinking was that by removing the unncessary await it avoids unnecessary microtasks.

**Review by codebytere — APPROVED** [review]
